message/messaging_service.cc

```diff
@@ -401,6 +401,17 @@ msg_addr messaging_service::get_source(const rpc::client_info& cinfo) {
 
 messaging_service::~messaging_service() = default;
 
+void messaging_service::set_my_host_id(locator::host_id id) {
+    _my_host_id = id;
+}
+
+locator::host_id messaging_service::get_my_host_id() {
+    if (!_my_host_id) {
+        on_internal_error(mlogger, "messaging_service::get_my_host_id: host ID missing");
```

Comment:
If not for internal error the whole function would just be `return _my_host_id.value()` :(

Reply:
But if the value is missing, `value()` would throw a cryptic exception that is super hard to debug. (We don't know where the exception originates from)